error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/util/SourceCode.java

```diff
@@ -24,4 +31,31 @@ public static String treeToString(Tree tree, VisitorState state) {
     String src = state.getSourceForNode(tree);
     return src != null ? src : tree.toString();
   }
+
+  /**
+   * Creates a {@link SuggestedFix} for the deletion of the given {@link Tree}, including any
+   * whitespace that follows it.
+   *
+   * <p>Removing trailing whitespace may prevent the introduction of an empty line at the start of a
+   * code block; such empty lines are not removed when formatting the code using Google Java Format.
+   *
+   * @param tree The AST node of interest.
+   * @param state A {@link VisitorState} describing the context in which the given {@link Tree} is
+   *     found.
+   * @return A non-{@code null} {@link SuggestedFix} similar to one produced by {@link
+   *     SuggestedFix#delete(Tree)}.
+   */
+  public static SuggestedFix deleteWithTrailingWhitespace(Tree tree, VisitorState state) {
+    CharSequence sourceCode = state.getSourceCode();
+    int endPos = state.getEndPosition(tree);
+    if (sourceCode == null || endPos == NOPOS) {
+      /* We can't identify the trailing whitespace; delete just the tree. */
+      return SuggestedFix.delete(tree);
+    }
+
+    int actualEnd = NON_WHITESPACE_MATCHER.indexIn(sourceCode, endPos);
+    return actualEnd == -1
+        ? SuggestedFix.delete(tree)
+        : SuggestedFix.replace(((DiagnosticPosition) tree).getStartPosition(), actualEnd, "");
```

Comment:
Not that I think we ever hit this case, but I think we should use `sourceCode.length()` if `actualEnd == -1`.

Reply:
Rebased and added a commit.